### ISSUE TITLE
localhost:3000/users just shows 'respond with a resource'

### ISSUE THREAD

**Issue by Thorstorm** [onedesign/express-socketio-tutorial#4]
Hello,
I was following your socket.io tutorial and I think I found a mistake. If you want to use your modified index.jade (with new socket.io code), you need to specify in './routes/users.js' the following:

    router.get('/', function (req, res, next) {
        res.io.emit('socketToMe', 'users');
        // res.send('respond with a resource')
        res.render('index');
    });
Am I right?

**Comment by danboy** [comment]
Good catch. I'll try and update today.

**Comment by Thorstorm** [comment]
Thank you. Tutorial was great, keep up the good work!

**Comment by danboy** [comment]
Hey Thorstrom-

Just looked at this again. The "user" routes are technically not being used in this example.

I intentionally left out cleaning them up so we wouldn't add confusion to the tutorial (it's not a necessary step).

Hope that didn't throw you off.

**Comment by Thorstorm** [comment]
At the end of your tutorial, you claim that typing `$ curl http://localhost:3000/users` will output at dev tool console the string `"users"`. I assume you refer to the browser's dev tools console, though. 

In order to output to the browser's dev tools console the string `"users"` sent from `.routes/users.js`, you need to send it to the modified **`.views/index.jade`** that looks like this: 

    extends layout

    block content
      h1= title
      p Welcome to #{title}
      script(src="/socket.io/socket.io.js")
      script.
        var socket = io('//localhost:3000');
        socket.on('socketToMe', function (data) {
          console.log(data);
        });

If you want the client-side socket listens to the `'socketToMe'` event and do the `console.log(data)`, where `data` will be `"users"`, you need to render **`.views/index.jade`**

Am I understanding the data flow correctly?

**Comment by danboy** [comment]
Nope, no need.

The index page (http:localhost:3000) is already registered to listen to events on socketToMe on the client side. The Users route is sending the message 'users' to that socket when the page is loaded.

In my demo you're never intended to visit the /users route in the browser, only hit it via curl from a terminal.

You could really put whatever you want in place of that 'users' string on line #6 in routes/users.js. Same with the string in the res.send block. that just completes the transaction.

Most commonly I'm sending down a data object via the socket.

```
router.get('/', function (req, res, next) {
    res.io.emit('socketToMe', {users: ['bob', 'john', 'thor']});
    res.send('respond with a resource')
});
```

Likely if we were going to have a user route render html it would be a different template than the index route.

tldr;

This demo intends you to: 
1. Open your browser to the index page (http://localhost:3000/)
2. Open dev tools to console.
3. open your terminal and type:

```
curl http://localhost:3000/users
```

**Comment by danboy** [comment]
Interesting I see where the confusion might be.

In the README I say "Now pop open another terminal" (I do my dev in VIM), I changed the to "Now pop open a terminal" so it hopefully doesn't confuse anyone else.

**Comment by Thorstorm** [comment]
So, you need to do `$curl http://localhost:3000/users` in a terminal in order to see on your http://localhost:3000 **browser's** dev console output the string `"users"`?

**Comment by danboy** [comment]
Well technically you just need to visit http://localhost:3000/users, but doing it from the terminal helps illustrate it similarly to another user visiting the page remotely.

So you could hit the page from your browser, another browser, or postman or whatever tool you like.

**Comment by Thorstorm** [comment]
I see, but the key thing here seems to be that wherever (terminal, browser, etc..) you request `http://localhost:3000/users` or `http://localhost:3000` from, you need to request first http://localhost:3000/ and then http://localhost:3000/users to see that http://localhost:3000 dev console output the string `users`, right?

**Comment by danboy** [comment]
That's correct. http://localhost:3000 is connected as a client socket, http://localhost:3000/users is passing info via a server socket.

**Comment by Thorstorm** [comment]
Good. Thank you for guide me
